### PR TITLE
Minor streamlining of simplifications and optimisations

### PR DESCRIPTION
**Pull request by PetarMax**
This PR streamlines the current simplifications and optimisations. Specifically, it:
- removes some `chop`-related simplifications that have been subsumed by other simplifications and `smt-lemma`s that are no longer required since booster can use Z3 while applying simplifications;
- tightens the wordstack-related constraints in optimisations, thereby minimising the number of simplifications required, especially when `bool2Word` terms are added;
- redefines `#rangeBool` as a range instead of a disjunction of possible values, in line with the other ranges.

### CODE REVIEW

**Review by anvacaru — DISMISSED** [review]

**Comment by ehildenb** [comment]
@PetarMax it looks ilke you need to rebase your work on the main branch, not on master.

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ module EVM-OPTIMIZATIONS
       ...
     </kevm>
     requires ( #if USEGAS #then Gverylow < SCHED > <=Gas GAVAIL #else true #fi )
-     andBool ( #sizeWordStack( chop( ( W0 +Int W1 ) ) : WS ) <=Int 1024 )
+     andBool ( #sizeWordStack ( WS ) <=Int 1022 )
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:194]: This doesn't look correct to me? It sohuld be 1023?

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md [hunk 2]
```diff
@@ -227,7 +227,7 @@ module EVM-OPTIMIZATIONS
       ...
     </kevm>
     requires ( #if USEGAS #then Gverylow < SCHED > <=Gas GAVAIL #else true #fi )
-     andBool ( #sizeWordStack( chop( ( W0 -Int W1 ) ) : WS ) <=Int 1024 )
+     andBool ( #sizeWordStack ( WS ) <=Int 1022 )
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:230]: Same here, 1023?

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md [hunk 3]
```diff
@@ -263,7 +263,7 @@ module EVM-OPTIMIZATIONS
       ...
     </kevm>
     requires ( #if USEGAS #then Gverylow < SCHED > <=Gas GAVAIL #else true #fi )
-     andBool ( #sizeWordStack( W0 &Int W1 : WS ) <=Int 1024 )
+     andBool ( #sizeWordStack ( WS ) <=Int 1022 )
```
Comment (ehildenb) [inline kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:266]: And here, 1023?

**Review by ehildenb — DISMISSED** [review]
Can you back out the partial evaluation of the optimizations? I tested out that optimization before, it wasn't significant.

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md (hunk 1 above)
Comment (PetarMax) [reply kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:194]: I think that it is correct, given that the size of the word stack is reducing and `W0 : W1 : WS` has to be of max size 1024?

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md (hunk 2 above)
Comment (PetarMax) [reply kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:230]: Same as above.

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md (hunk 3 above)
Comment (PetarMax) [reply kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:266]: Same as above.

**Comment by PetarMax** [comment]
> Can you back out the partial evaluation of the optimizations? I tested out that optimization before, it wasn't significant.

While testing engagement code with @geo2a, at some point we were stuck trying to simplify the`bool2Word`s in the `requires` clause. They are not needed there, because the computation is on the size of the stack, not the contents.

**Comment by PetarMax** [comment]
> @PetarMax it looks ilke you need to rebase your work on the main branch, not on master.

There's no `main` branch - this branch is a bit of a Frankenstein, since I used to merge and rebase off of a lot of versions of master before. I'll see what I can do.

**Comment by PetarMax** [comment]
I've tried to do `git rebase -i master` and then choose only my commits, but I ended up with no changes. This works when I accidentally merge `master` instead of `develop` in `pyk`...

**Comment by ehildenb** [comment]
@PetarMax if the optimization changes are made, please make them have the correct wordstack size limits. I see 1022 used in a lot of places that it should be 1023 I think.

**Comment by ehildenb** [comment]
I re-opened my PR for partial evaluation of the optimizations, we can probably get that merged separately. It was actually inspired by Pi2 work: https://github.com/runtimeverification/evm-semantics/pull/2566

**Comment by PetarMax** [comment]
> @PetarMax if the optimization changes are made, please make them have the correct wordstack size limits. I see 1022 used in a lot of places that it should be 1023 I think.

I have answered this [here](https://github.com/runtimeverification/evm-semantics/pull/2565#discussion_r1707231354), I don't believe that 1023 is correct given that the word stack on the LHS then be of size 1025.

**Comment by PetarMax** [comment]
> I re-opened my PR for partial evaluation of the optimizations, we can probably get that merged separately. It was actually inspired by Pi2 work: #2566

Yes, very happy to merge that one.

**Pull request closed without merging**